### ISSUE TITLE
Finetune Problem

### ISSUE THREAD

**Issue by ChinnYu** [smallcloudai/refact#235]
I wanted to extend my sincere appreciation for your project. As a devoted fan of your work, I have thoroughly enjoyed being part of this journey. However, I have recently encountered an issue with the latest version. Specifically, when attempting to perform code finetuning and clicking on "Run Filter," it displays a fail message. However, when I click on "Run Finetune," it shows that the process is completed successfully, and the code even indicates acceptance. On top of that, the run finetune process gets interrupted immediately. This problem did not exist in the previous version.
![MicrosoftTeams-image (9)](https://github.com/smallcloudai/refact/assets/32726550/8c5c6441-7934-44e0-b3e0-07b3ec61d39b)
![MicrosoftTeams-image (10)](https://github.com/smallcloudai/refact/assets/32726550/579d1377-4cda-41ea-934f-d678d86136f5)
![image](https://github.com/smallcloudai/refact/assets/32726550/700b503b-8508-41d8-aa9e-1dfb0f888a1f)

I understand that software development can be complex, and unforeseen issues may arise. Therefore, I kindly request your assistance in resolving this matter. If there are any potential solutions or suggestions you could provide to prevent the interruption during the run finetune process, it would be greatly appreciated.

Once again, thank you for your dedication and hard work. I eagerly anticipate future updates and improvements to this remarkable project.

Thank you sincerely,

**Comment by ChinnYu** [comment]
Hello, @hazratisulton.  I used the source code and the provided Dockerfile to build the image. I noticed that the built image has an error when I press 'run filter.
![image](https://github.com/smallcloudai/refact/assets/32726550/b2f628e3-ceed-4c24-881d-77a5327bb643)

**Comment by JegernOUTT** [comment]
@hazratisulton have you managed to reproduce it?

**Comment by hazratisulton** [comment]
> @hazratisulton have you managed to reproduce it?

No, I couldn't. I asked @mitya52 to take a look, maybe he could offer some ideas.

**Comment by ChinnYu** [comment]
HI @hazratisulton @JegernOUTT , I attempted to build the image using the latest version of the source code (12/28) from the 'dev' branch, and it seems that the same issue persists. If there's a specific log for analysis that you need, please let me know, and I'll provide it.

![image](https://github.com/smallcloudai/refact/assets/32726550/19e260d0-1c75-44ec-bd4c-8032f700e88c)

**Comment by ChinnYu** [comment]
After numerous code modifications, I discovered that changing '_aux' to 'aux' allows locating the Python module. But it also brings about two issues. 1. 'Index out of bounds' occurs when pressing 'run filter' and selecting 'codellama.' 2. 'AssertionError: You have to have more files to process than processes' happens at the beginning of Finetune. Indeed, the number of my files exceeds the number of processes. The first one is resolved by changing to transformers==4.34.0, and for the second one, the allocation rules need to be modified.

![image](https://github.com/smallcloudai/refact/assets/32726550/91e9b0d2-c159-4f5e-a644-d43742a53266)

![image](https://github.com/smallcloudai/refact/assets/32726550/15a9feb9-9b8f-4bff-a912-e19c3873311f)

**Comment by olegklimov** [comment]
Interesting! But it works in nightly without any changes 🤔   Let's ask what @JegernOUTT and @mitya52 think.

**Comment by ChinnYu** [comment]
Hi, I'd like to ask another question. I'm attempting to integrate 'deepseek-ai/deepseek-coder-33b-base' into the refact. I added the 33B model to these two files: 'refact/known_models_db/refact_known_models/huggingface.py' and 'self_hosting_machinery/finetune/configuration/supported_models.py'. The modification process is similar to 'deepseek-coder/5.7b/mqa-base', and I've also added 'known_models' in 'refact-lsp'. However, Visual Studio Code (vscode) continues to report the following errors. I have checked the main page and confirmed that the model has been successfully initialized. Do the experts have any debugging suggestions for this issue? 
![image](https://github.com/smallcloudai/refact/assets/32726550/6ba1220f-d971-417b-b1cc-448c859a17a2)

Additionally, there is a warning as shown in the second image. What should be configured in this case? Thank you.

![image](https://github.com/smallcloudai/refact/assets/32726550/973a983a-a724-4294-865f-9b911cd9beca)

**Comment by olegklimov** [comment]
@ChinnYu awesome that you are trying this! You might need a change in `refact-lsp`, just add the model there by analogy like the other models.

There was this idea to try new models using "works like this other known model" in settings. But then it appeared not very practical (the best settings is no settings, because it gets outdated, needs tech support to remove unnecessary settings once it's there and server side changes, etc). Or maybe we could return to this idea, because it allows to try a model quickly without recompiling the lsp.

**Comment by JegernOUTT** [comment]
```
 I discovered that changing 'aux' to '_aux' allows locating the Python module
```
It's interesting, we never had such import namings, check this out 
https://github.com/smallcloudai/refact/blob/main/self_hosting_machinery/finetune/scripts/finetune_filter.py#L16
Are you sure you haven't changed them yourself accidentally?

**Comment by ChinnYu** [comment]
HI @JegernOUTT  I obtained the GitHub code by downloading the zip file, and I'm using 7zip as the decompression software. It seems that the built-in zip tool in Windows cannot decompress it. I've tried several times, but it keeps generating '_aux', as shown in the figure below.

![image](https://github.com/smallcloudai/refact/assets/32726550/9ac93f49-8096-435d-9509-b34a737cae48)

**Comment by JegernOUTT** [comment]
@ChinnYu 
Sorry for the delay, completely forgot about this
Yes, I've got the problem, it's due to some legacy Windows folder naming limitations.
We'll rename those folders in the next release, sorry for the inconvenience

**Comment by hazratisulton** [comment]
Given that this issue was reported over two years ago and there has been no recent activity, we are closing it as part of our issue tracker maintenance. If you (or anyone else) are still experiencing this problem on the latest release, please feel free to comment here with updated information (e.g., current version, steps to reproduce), and we will be happy to reopen it. Thank you for your understanding.
